src/web4/handler.rs

```diff
@@ -91,73 +89,105 @@ pub fn web4_get(contract: &Contract, request: Web4Request) -> Web4Response {
             } else {
                 title = " - Proposals".to_string();
             }
-            redirect_path =
-                format!("{}/widget/app?page={}&id={}", &current_account_id, page, id);
+            redirect_path = format!("{}/widget/app?page={}&id={}", &current_account_id, page, id);
             initial_props_json = json!({"page": page, "id": id});
         }
+        // Handle blog route with community and blog title
+        ("blog", Some(community), Some(last)) => {
+            // Split the query parameters from the blog title
+            let blog_title = last.split('?').next().unwrap();
+
+            redirect_path = format!(
+                "{}/widget/app?page=blogv2&community={}&id={}",
+                &current_account_id, community, blog_title
```

Comment:
Have you tested it with titles that have spaces, question marks, ampersands, and slashes? Please, add tests for these cases